File: verifier-problems/src/main/java/com/jetbrains/pluginverifier/utils/LocationUtils.java
package com.jetbrains.pluginverifier.utils;

import org.jetbrains.annotations.NotNull;
import org.objectweb.asm.tree.ClassNode;
import org.objectweb.asm.tree.FieldNode;
import org.objectweb.asm.tree.MethodNode;

/**
 * @author Sergey Patrikeev
 */
public class LocationUtils {

  @NotNull
  public static String getMethodLocation(@NotNull ClassNode classNode, @NotNull MethodNode methodNode) {
    return getMethodLocation(classNode.name, methodNode);
  }

  @NotNull
  public static String getMethodLocation(@NotNull String ownerClassName, @NotNull String methodName, @NotNull String methodDesc) {
    //NotNull checks are performed automatically
    return ownerClassName + '#' + methodName + methodDesc;
  }

  @NotNull
  public static String getMethodLocation(@NotNull String ownerClassName, @NotNull MethodNode methodNode) {
    return getMethodLocation(ownerClassName, methodNode.name, methodNode.desc);
  }

  @NotNull
  public static String getFieldLocation(@NotNull String ownerClassName, @NotNull String fieldName, @NotNull String fieldDescriptor) {
    //NotNull checks are performed automatically
    return ownerClassName + '#' + fieldName + "#" + fieldDescriptor;
  }

  @NotNull
  public static String getFieldLocation(@NotNull String ownerClassName, @NotNull FieldNode fieldNode) {
    return getFieldLocation(ownerClassName, fieldNode.name, fieldNode.desc);
  }
}

File: verifier-problems/src/main/java/com/jetbrains/pluginverifier/utils/LocationUtils.kt
package com.jetbrains.pluginverifier.utils

import org.objectweb.asm.tree.ClassNode
import org.objectweb.asm.tree.FieldNode
import org.objectweb.asm.tree.MethodNode

/**
 * @author Sergey Patrikeev
 */
object LocationUtils {

  fun getMethodLocation(classNode: ClassNode, methodNode: MethodNode): String {
    return getMethodLocation(classNode.name, methodNode)
  }

  fun getMethodLocation(ownerClassName: String, methodName: String, methodDesc: String): String {
    //NotNull checks are performed automatically
    return ownerClassName + '#' + methodName + methodDesc
  }

  fun getMethodLocation(owner: ClassNode, methodName: String, methodDesc: String): String {
    return getMethodLocation(owner.name, methodName, methodDesc)
  }

  fun getMethodLocation(ownerClassName: String, methodNode: MethodNode): String {
    return getMethodLocation(ownerClassName, methodNode.name, methodNode.desc)
  }

  fun getFieldLocation(owner: ClassNode, fieldName: String, fieldDescriptor: String): String {
    return getFieldLocation(owner.name, fieldName, fieldDescriptor)
  }

  fun getFieldLocation(ownerClassName: String, fieldName: String, fieldDescriptor: String): String {
    //NotNull checks are performed automatically
    return "$ownerClassName#$fieldName#$fieldDescriptor"
  }

  fun getFieldLocation(ownerClassName: String, fieldNode: FieldNode): String {
    return getFieldLocation(ownerClassName, fieldNode.name, fieldNode.desc)
  }
}
